llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -381,12 +465,39 @@ bool InterBlockScheduling::updateFixPoint(BlockState &BS) {
     // Iterate on CurMBB
     return false;
   }
+  BS.setScheduled();
+
+  int &II = BS.FixPoint.II;
+  if (II) {
+    if (BS.isPipelined()) {
+      return true;
+    }
 
+    // We cut off at the load latency because the main benefit from
+    // post-pipelining comes from wrapping latencies over multiple stages.
+    // This will only work if the stages are small enough. If we miss
+    // opportunities because of this, we should try to get the Prepipeliner
+    // to accept it.
+    if (II >= 7) {
+      // Fall back to the loop schedule. Note that we can only have II != 0
+      // after the loop schedule has stabilized.
+      BS.setPipeliningFailed();
+      return true;
+    }
+    II++;
+    return false;
+  }
   DEBUG_LOOPAWARE(dbgs() << "Converged,"
                          << " LatencyExtent=" << BS.FixPoint.MaxLatencyExtent
                          << " ResourceExtent=" << BS.FixPoint.MaxResourceExtent
                          << "\n");
 
+  // The loop schedule has converged, so we could declare our work done.
+  // But first try SWP
+  if (BS.getRegions().size() == 1 && BS.getPostSWP().accept(*BS.TheBlock)) {
+    II = 1;
+    return false;
```

Comment:
Once isScheduled is reached, the regular schedule is fixed, it can only be undone by a succesful postpipelining, which rebuilds the MachineInstr order from scratch. Currently we need one thing from the linear schedule, which is its length, since that determines the number of copies when creating the unrolled DAG. I've been doubting that, since it holds the assumption that this is on par with what the postpipeliner will find. 
The cleanest way would be to generate the nromal single copy DAG, then run scheduleFirstIteration, then use the length from there as stagecount to add the necessary copies to the DAG, then run scheduleOtherIterations.
At this point that would be quite invasive.